File: js/169. Majority Element.js
var majorityElement = function(nums) {
    let elementMap = {};
    let higherElement = nums[0];
    let highest = 1;
    for(const element of nums){
        elementMap[element] = (elementMap[element] || 0) + 1;
        if (elementMap[element] > highest) {
            highest = elementMap[element];
            higherElement = element;
        }
    }
    return higherElement;
};

File: python/leetcode/169. Majority Element.py
def majorityElement(self, nums):
        elementMap = {}
        higherElement = nums[0]
        highest = 1
        for x in nums:
            elementMap[x] = elementMap.get(x, 0) + 1
            if elementMap[x] > highest:
                highest = elementMap[x]
                higherElement = x

        return higherElement